pandas/core/groupby/groupby.py

```diff
@@ -2412,6 +2415,37 @@ def _get_cythonized_result(
         output: Dict[base.OutputKey, np.ndarray] = {}
         base_func = getattr(libgroupby, how)
 
+        if how == "group_quantile":
```

Comment:
Hmm I realize we have a few patterns like this already but not ideal, as we want to try and keep these functions that interact with Cython as generic as possible. What's the motivation for doing this this way?

Reply:
This would not be in the final product, I only segmented it off so all tests would pass. I realize now it's a bit confusing, so I went ahead and rewrote _get_cythonized_result as it really wasn't much effort (but could probably be improved). Hopefully it's more clear now. The next step would be to rewrite the three other cython implementations so that their signature agrees with what's now expected in _get_cythonized_result.

Reply:
Also, I wanted to make sure this is understood: if _get_cythonized_result is changed to what's in this PR, then we can use it on a whole host of cythonized functions that wouldn't work today. In particular, we'd be able to use it on group_var, which would eliminate the branching you identified back in #33630.